sda-download/api/sda/sda.go

```diff
@@ -272,23 +272,18 @@ func Download(c *gin.Context) {
 		return
 	}
 
-	switch c.Param("type") {
-	case "encrypted":
-		// calculate coordinates
-		start, end, err = calculateEncryptedCoords(start, end, c.GetHeader("Range"), fileDetails)
-		if err != nil {
-			log.Errorf("Byte range coordinates invalid! %v", err)
+	wholeFile := true
+	if start != 0 || end != 0 {
+		wholeFile = false
+	}
```

Comment:
This entire wholeFile thingy could be removed if the check inside the switch case looks something like this:
```
		if start == 0 && end == 0 {
			file, err := Backend.NewFileReader(fileDetails.ArchivePath)
			if err != nil {
				log.Errorf("could not find archive file %s, %s", fileDetails.ArchivePath, err)
				c.String(http.StatusInternalServerError, "archive error")
		
				return
			}

			fileStream = io.MultiReader(newHr, file)
		} else {
			file, err := Backend.NewFileReadSeeker(fileDetails.ArchivePath)
```

Reply:
Yes, but the end position might have been altered in `calculateCoords`. So one needs to make sure to check the original positions for the `wholeFile` check, if doing this kind of refactoring.

Reply:
to be picked up in #834.